email/add_ecr/email_sender.py

```diff
@@ -0,0 +1,311 @@
+              <td align="left" valign="top" style="padding: 10px; font-family: Arial, sans-serif;">
+              <h3 style="margin: 0 0 10px 0; font-size: 18px; color: #2c3e50;">{author_name}'s engagement:</h3>
+              <p style="margin: 0; font-size: 14px; color: #555;"><p style="margin:5px 0 0 0;">
+              <strong>Daily Shelved:</strong> {daily_shelved} <br> 
+              <strong>Avg Rating Change:</strong> {avg_rating_change}
+              </p>
+            </td>
+          </tr>
+        </table>              
+        '''
+    return container_html
+
+
+def generate_html_head():
+    '''Generates the html head for the email'''
+    html = '''
+    <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Publisher Report Email</title>
+    </head>
+    '''
+    return html
+
+
+def generate_html_body(publisher_id: int) -> str:
+    '''Returns the html body of the email message'''
+    publisher_name = get_publishers_name(publisher_id)
+    if not publisher_name:
+        raise ValueError('No valid publisher for the given id.')
+
+    author_ids = get_publishers_tracked_authors(publisher_id)
+    author_ids = [(1,)]
+    if not author_ids:
+        html = f'''<body style="font-family: 'Poppins', Arial, sans-serif; background-color: #f9f9f9; padding: 20px; color: #333333; line-height: 1.5;"> 
+        Dear {publisher_name}, 
+        <br><br>
+        No tracked authors were found. Please add an author on the Bookworm dashboard.'''
+        return html + '</body>'
+
+    html_cards = ""
+    for id in author_ids:
+        id = id[0]
+        html_cards = html_cards+generate_author_html_container(id)
+
+    html = f'''
+    <body style="font-family: 'Poppins', Arial, sans-serif; background-color: #f9f9f9; padding: 20px; color: #333333; line-height: 1.5;>
+      <table width="100%" cellpadding="0" cellspacing="0" border="0" style="max-width: 600px; margin: 0 auto; background-color: #ffffff; border: 1px solid #e0e0e0; padding: 20px;">
+        <tr>
+          <td>
+            <p style="font-size: 16px; margin: 0 0 15px 0;">Dear <strong>{publisher_name}</strong>,</p>
+            <p style="font-size: 15px; margin: 0 0 20px 0;">The following authors had noteworthy engagement:</p>
+      <table width="100%" border="0" cellspacing="0" cellpadding="0" style="background-color: #fafafa; border-radius: 6px; overflow: hidden;">
+    '''
+    html = html + html_cards
+    return html + '</body>'
+
+
+def format_html_email(head: str, body: str) -> str:
+    '''Formats the html for an update email'''
+    return '<html lang="en">' + head + body + '</html>'
+
+
+def get_email_subject(publisher_id: int) -> str:
+    '''Formats and returns the email subject for a given publisher id'''
+    date = get_todays_date_formatted()
+    name = get_publishers_name(publisher_id)
+    return f'{name}\'s Daily Tracker {date}'
+
+
+def aws_send_email(email_html: str, publisher_id: int):
```

Comment:
Missing return type hint here
```suggestion
def aws_send_email(email_html: str, publisher_id: int) -> None:
```

Reply:
commit this locally